### ISSUE TITLE
[HOLD for payment 2024-01-17] [$500] LHN - Red dot does not appear in LHN conversation when IOU deletion error occurs

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33535]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.16-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open New Expensify app
2. Log in with any account
3. Navigate to any conversation with an user you have access to and Request $10
4. Disable the internet connection on the main testing device.
5. As the secondary user/device navigate to the conversation with the main user where the 10$ were just requested
6. As the Secondary user/device Pay the IOU via "Pay elsewhere"
7. On the main testing device while offline - cancel the 10$ IOU
8. Enable the internet connection on the main testing device
9. Navigate to a paid IOU

## Expected Result:
The red dot should appear in the LHN conversation when the error is caused by an offline deletion of a paid IOU

## Actual Result:
A red dot does not appear in the LHN conversation when the error is caused by a paid IOU being removed offline

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/2c3b5833-437e-4e9b-bc06-98238b40b510

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c5cf5023e3aab261</li>
        <li>Upwork Job ID: 1738392838951575552</li>
        <li>Last Price Increase: 2023-12-23</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 28071466</li>
<li>tienifr | Contributor | 28071467</li>
</ul></li>
    </ul>
</details>

**Comment by shubham1206agra** [comment]
@tienifr Your proposal doesn't seem right in this case. Your RCA is correct but solution seems wrong

**Comment by brunovjk** [comment]
## Proposal

### Problem Statement
The red dot isn't appearing in the Expensify LHN conversation when a paid IOU is removed offline.

### Root Cause
During the deletion of a money request, the `IOU.deleteMoneyRequest.failureData` doesn't set the error message to the `reportPreviewAction` in Onyx.

### Solution
To resolve this, we propose enhancing the code responsible for updating `ONYXKEYS.COLLECTION.REPORT_ACTIONS` for `chatReport`. Instead of a simple merge, explicitly set the errors field for the `reportPreviewAction`. Also, modify `ReportActions.clearReportActionErrors` to clear errors in both the IOU report and its parent chat.

### Alternative Solutions Explored
Alternatively, making the IOU report 'unread' when an error emerge, could push it to the top of the list upon reconnection, eliminating the need to send the error message in two places.

**Comment by shubham1206agra** [comment]
> > @tienifr Your proposal doesn't seem right in this case. Your RCA is correct but solution seems wrong
> 
> @shubham1206agra just curious what do you think seems wrong about it?
> 
> Since we're doing the same thing when creating money request [here](https://github.com/Expensify/App/blob/b74be723548d45d47dde7fe29aff5eafe16b290f/src/libs/actions/IOU.js#L518C43-L518C43). That's why creating money request doesn't have the same problem.

Oh shoot, I actually got confused with something else. Your solution is correct. Sorry for the confusion.

> Currently even for other money request-related errors (like when creating money request fails), if we want to clear error by clicking "X", we have to clear errors individually in the report preview action and the IOU report. This will be the same for the delete money request error here.

Lets keep this as it is right now as it is out of scope for this PR.

**Comment by shubham1206agra** [comment]
@tienifr's [proposal](https://github.com/Expensify/App/issues/33535#issuecomment-1868216069) looks good to me

🎀👀🎀 C+ reviewed

**Comment by robertjchen** [comment]
Agreed, let's go with @tienifr 's proposal!

**Comment by robertjchen** [comment]
Awaiting completion of BZ checklist 🙏

**Comment by shubham1206agra** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@shubham1206agra] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/17583
* [x]  [@shubham1206agra] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/17583/files#r1463432049
* [x]  [@shubham1206agra] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not required
* [x]  [@shubham1206agra] Determine if we should create a regression test for this bug. No
* [x]  [@shubham1206agra] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@Christinadobrzyn] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Christinadobrzyn** [comment]
Sorry for the delay- paying this out. 

Payouts due:

Issue Reporter: NA
Contributor: $500 @tienifr (paid in Upwork)
Contributor+: $500 @shubham1206agra (in Upwork)

Eligible for 50% #urgency bonus? NA

Upwork job is [here](https://www.upwork.com/jobs/~01c5cf5023e3aab261).

@shubham1206agra can you accept our offer in Upwork and I'll pay you? Thanks!

**Comment by shubham1206agra** [comment]
Accepted

**Comment by Christinadobrzyn** [comment]
Awesome! paid in Upwork based on this payment summary - https://github.com/Expensify/App/issues/33535#issuecomment-1906852954

Closing this out as complete! Thanks everyone!
